File: src/main/webapp/service/wareService.js
/**
 * Created by JELENA on 3.6.2017.
 */

app.service('wareService', function($http) {
   return {
       read: function(onSuccess, onError) {
           $http.get('/api/wares').then(onSuccess, onError);
       },
       create: function(ware, onSuccess, onError) {
           $http.post('/api/wares', ware).then(onSuccess, onError);
       },
       update: function(ware, onSuccess, onError) {
           $http.patch('/api/wares/' + ware.id, ware).then(onSuccess, onError);
       },
       delete: function(id, onSuccess, onError) {
           $http.delete('/api/wares/' + id).then(onSuccess, onError);
       }
   }
});

File: src/main/java/com/ftn/service/WareService.java
package com.ftn.service;

import com.ftn.model.Ware;

import java.util.List;

/**
 * Created by JELENA on 30.5.2017.
 */
public interface WareService {

    List<Ware> read();

    Ware create(Ware ware);

    Ware update(Long id, Ware ware);

    void delete(Long id);
}
